scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
+    "\n",
+    "for var_name in ds.keys():\n",
+    "    stitch_data(var_name)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Adds a coordinate for the spectra\n",
-    "ds.coords[Dim.Spectrum] = sc.Variable([Dim.Spectrum], values=np.arange(ds[\"sample\"].shape[1]))\n",
-    "ds"
+    "integrated = sc.Dataset()\n",
+    "for dim in stitched.coords.keys():\n",
+    "    integrated.coords[dim] = stitched.coords[dim]\n",
+    "\n",
+    "integrated = sc.sum(stitched, 'tof')\n",
+    "# Put TOF coords back for the export to TIFF stack later\n",
+    "integrated.coords[\"tof\"] = stitched.coords[\"tof\"]"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# Pull out reference from out dataset to avoid checking for it in loops\n",
+    "reference = stitched[\"reference\"]\n",
+    "integrated_reference = integrated[\"reference\"]\n",
+    "\n",
+    "# Causes segmentation fault, see scipp/scipp issue: #1176\n",
+    "# when this can be added back we can remove the \"if \"reference\" checks later\n",
+    "# del stitched[\"reference\"]\n",
+    "# del integrated[\"reference\"]\n"
+   ],
    "metadata": {
-    "scrolled": true
-   },
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
    "outputs": [],
    "source": [
-    "stitched = sc.Dataset(coords={Dim.Tof: sc.Variable([Dim.Tof], values=np.arange(*rebin_parameters, dtype=np.float64))})\n",
-    "\n",
-    "stitched[\"sample\"] = stitch(ds[\"sample\"], frame_parameters, rebin_parameters)\n",
-    "stitched[\"reference\"] = stitch(ds[\"reference\"], frame_parameters, rebin_parameters)\n",
-    "\n",
-    "stitched"
-   ]
+    "num_spectra = stitched.coords[\"spectrum\"].shape[0]\n",
+    "bank_width = int(np.sqrt(num_spectra))\n",
+    "\n",
+    "def mask_non_sample_regions(integrated_dataset, stitched_dataset, var_name):\n",
+    "    integrated_spectra = integrated_dataset[var_name]\n",
+    "\n",
+    "    # This should use sc.greater once #1178 is completed\n",
+    "    spectra_masks = np.greater(integrated_spectra.values, masking_threshold)\n",
+    "    spectra_masks = sc.Variable([\"spectrum\"], values=spectra_masks)\n",
+    "\n",
+    "    # Some pixels may be noisy / not strong enough signal so they erranously\n",
+    "    # become unmasked / masked from the above operation. This step checks if *all*\n",
+    "    # the surrounding pixels are True or False then sets the center pixel to\n",
+    "    # that value if they are, removing \"salt and pepper\" noise\n",
+    "    final_mask = operations.mask_from_adj_pixels(spectra_masks, bank_width)\n",
+    "    stitched_dataset.masks[var_name] = final_mask\n",
```

Comment:
This is a very Mantidy way of writing things. I suggest to `return final_mask` and change the function signature to:

```python
def mask_non_sample_regions(integrated_spectra):
```
for much more readable and understandable code on the calling site:
```python
stitched[k] = mask_non_sample_regions(integrated[k])
```

Reply:
Ahh yes Owen commented on the use of globals too and I went through and fixed the ones I found from the original but this slipped past :+1: